numba/core/ir_utils.py

```diff
@@ -1171,26 +1171,46 @@ def dprint_func_ir(func_ir, title, blocks=None):
 
 def find_topo_order(blocks, cfg = None):
     """find topological order of blocks such that true branches are visited
-    first (e.g. for_break test in test_dataflow).
+    first (e.g. for_break test in test_dataflow). This is written as an iterative
+    implementation of post order traversal to avoid recursion limit issues.
     """
     if cfg is None:
         cfg = compute_cfg_from_blocks(blocks)
+
     post_order = []
+    # Has the node already added its children?
     seen = set()
-
-    def _dfs_rec(node):
-        if node not in seen:
+    # Has the node already been pushed to post order?
+    visited = set()
+    stack = [cfg.entry_point()]
+
+    while len(stack) > 0:
+        node = stack[-1]
+        if node not in visited and node not in seen:
+            # We haven't added a node or its children.
             seen.add(node)
             succs = cfg._succs[node]
             last_inst = blocks[node].body[-1]
             if isinstance(last_inst, ir.Branch):
-                succs = [last_inst.falsebr, last_inst.truebr]
+                succs = [last_inst.truebr, last_inst.falsebr]
```

Comment:
Why does the order of `truebr` `falsebr` matter, curiously?

Reply:
This is just to maintain the same order as the recursive implementation in case anything depended on it (as the stack effectively reverses the order in which the `succs` are visited. I do not believe this should matter, but I was just trying to be conservative to limit changes in case anything depended on the relative ordering of the true and false label. Happy to revert this if you would like.